src/ctap/mod.rs

```diff
@@ -212,45 +211,55 @@ fn truncate_to_char_boundary(s: &str, mut max: usize) -> &str {
 
 /// Parses the metadata of an upgrade, and checks its correctness.
 ///
-/// Returns the hash over the upgrade, including partition and some metadata.
 /// The metadata consists of:
-/// - 32B upgrade hash (SHA256)
-/// -  4B timestamp (little endian encoding)
-/// -  4B partition address (little endian encoding)
-/// The upgrade hash is computed over the firmware image and all metadata,
-/// except the hash itself.
+/// - 32 B upgrade hash (SHA256)
+/// - 64 B signature,
+/// - 32 B padding,
+/// - 20 B version and
+/// -  4 B partition address in little endian encoding.
+/// Checks the hash and whether the signature is correct.
 fn parse_metadata(
     upgrade_locations: &impl UpgradeStorage,
+    public_key_bytes: &[u8],
     metadata: &[u8],
-) -> Result<[u8; 32], Ctap2StatusCode> {
-    const METADATA_LEN: usize = 40;
+) -> Result<(), Ctap2StatusCode> {
+    const METADATA_LEN: usize = 0x1000;
     if metadata.len() != METADATA_LEN {
         return Err(Ctap2StatusCode::CTAP1_ERR_INVALID_PARAMETER);
     }
+    let metadata_address = LittleEndian::read_u32(&metadata[148..][..4]);
```

Comment:
Shouldn't we ideally use constants for offsets in the metadata? (0, 32, 140, 148)

Reply:
I kept some numbers that only matter inside the function that uses them. I can also make them const, if you like. I went with your suggestion to use half pages.